resources/indexer.py

```diff
@@ -339,53 +532,26 @@ def process_dir(path, site):
     dirs = []
     file_meta = []
 
-    # get directory's metadata
-    run_meta_xml = None
-    gaps_files = dict()  # gaps_files[<filename w/o extension>]
-    gcd_files = dict()  # gcd_files[<run id w/o leading zeros>]
-    for dir_entry in scan:
-        if "meta.xml" in dir_entry.name:  # Ex. level2_meta.xml
-            with open(dir_entry.path, 'r') as f:
-                run_meta_xml = xmltodict.parse(f.read())
-        if "_GapsTxt.tar" in dir_entry.name:  # Ex. Run00130484_GapsTxt.tar
-            with tarfile.open(dir_entry.path) as tar:
-                for tar_obj in tar:
-                    file = tar.extractfile(tar_obj)
-                    file_dict = yaml.load(file)
-                    # Ex. Level2_IC86.2017_data_Run00130484_Subrun00000000_00000188_gaps.txt
-                    name = tar_obj.name.split("_gaps.txt")[0]
-                    gaps_files[name] = file_dict
-        if "GCD" in dir_entry.name:  # Ex. Level2_IC86.2017_data_Run00130484_0101_71_375_GCD.i3.zst
-            run = _get_run_number(dir_entry)
-            gcd_files[str(run)] = dir_entry.path
+    manager = MetadataManager(path, site)
```

Comment:
Question: what happens when a directory isn't labelled correctly, but the files all start with PFRaw?  Does the manager assign the wrong level, and metadata is missed?

I'm wondering because of the loading that occurs off of disk every year, or other similar circumstances.

Reply:
This is a requirement, the script would exit if the directory doesn't contain a processing level.

Is this problematic?

Reply:
@jnbellinger how do you ingest files from disk?  Are you planning on using this script?

Reply:
I use "ltacmd catalog load" for each directory that I've identified as full.  I'm not sure if this gets called behind the scenes:  I don't see it in the LTA code.

Reply:
Checking LTA code, it looks like it does not call this.  Instead it does the bare minimum (just getting the checksum) and puts it directly in the file catalog.  I'm not sure if it's good or bad that those files don't have any other metadata.

Reply:
It is a bit generic.  For raw data, what extra benefit to we get from the metadata that isn't already encoded in the "PFRAw" and run number and whatever live tells us about the data quality? The beginning and end events; what else?

Reply:
I mean, in the LTA code you don't even know that it's a raw file, unless you look at the name.  This script at least adds fields for the processing level, run, subrun, file part, beginning and ending events, (maybe others?).

That's important if you're filtering for, say, all files that have to do with a run.  It's just not possible to look at the name of every file in the catalog and ask if some string is in the name (it would take forever, because there isn't a good way to index that query).

Reply:
I suppose one could do "after the fact" fixup for some of those things.  Or write something that simultaneously checksums and unpacks/reads the metadata file.  Doing a second read pass during the "catalog load" to pull the metadata and read it seems excessive.

Is the metadata available anywhere other than the bundled metadata file?

Reply:
Is there a resolution here for the directory naming scheme?

I chose to look at the directory path name instead of the file name to handle older files like `/data/exp/IceCube/2008/filtered/level2/0324/PFFilt_PhysicsTrig_PhysicsFiltering_Run00110714_Level2_Part00000034.i3.gz`.

Reply:
@blinkdog any thoughts on this?

Reply:
On the issue of metadata in the catalog, `ltacmd catalog load` should probably be deprecated in favor of this script after it is merged. I needed a quick way for the operator to get entries in the catalog for production deployment testing, but really there's no substitute for doing it right the the first time. This script is 'doing it right' in terms of catalog metadata.

Reply:
On the issue of file identification, I would look simply for `PFRaw`, `PFFilt`, and `PFDST` anywhere within the full path. There should be no data files anywhere that have multiple tags like this in the path. A case like that would mean the file is in the wrong directory, or the file has a _very_ strange name (PFRaw_PFFilt_blah_blah), and both are something we'd probably like an operator to correct before we start filling the catalog with metadata about them.

As for L2, L3, L4, I'm not really familiar with the unique substring tag that would identify a file of that type. I imagine there's got to be one for each, but I haven't worked with those files very often, so I'm in the dark.

Reply:
I think we should leave as-is for this pull request, but think of this for a new pull request.

Reply:
I've not seen a unique substring tag for L2/L3/L4 either.  
And I agree that parsing the directory and file name is probably good enough.  Every now and then we got duplicate run numbers.  In the past this hasn't been a problem, since one or more of the duplicates were test runs.  I assume that will continue to be the reason for such glitches in the future, and thus not a serious problem.
Will we want start/stop times first/last  event numbers in the catalog too?  Once upon a time I thought that might be good to have for coincidence studies with other detectors, but after seeing the time windows they use, I suspect just having the right run number might be good enough for that work.